src/main/java/net/neoforged/neoforge/client/NeoForgeRenderTypes.java

```diff
@@ -158,198 +155,203 @@ public RenderType get() {
+            var rendertype$state = RenderType.CompositeState.builder()
+                    .setShaderState(RenderType.RENDERTYPE_TEXT_INTENSITY_SEE_THROUGH_SHADER)
                     .setTextureState(new CustomizableTextureState(locationIn, () -> NeoForgeRenderTypes.enableTextTextureLinearFiltering, () -> false))
-                    .setTransparencyState(TRANSLUCENT_TRANSPARENCY)
-                    .setLightmapState(LIGHTMAP)
-                    .setDepthTestState(NO_DEPTH_TEST)
-                    .setWriteMaskState(COLOR_WRITE)
+                    .setTransparencyState(RenderType.TRANSLUCENT_TRANSPARENCY)
+                    .setLightmapState(RenderType.LIGHTMAP)
+                    .setDepthTestState(RenderType.NO_DEPTH_TEST)
+                    .setWriteMaskState(RenderType.COLOR_WRITE)
                     .createCompositeState(false);
-            return create("neoforge_text_see_through", DefaultVertexFormat.POSITION_COLOR_TEX_LIGHTMAP, VertexFormat.Mode.QUADS, 256, false, true, rendertype$state);
+            return RenderType.create("neoforge_text_see_through", DefaultVertexFormat.POSITION_COLOR_TEX_LIGHTMAP, VertexFormat.Mode.QUADS, 256, false, true, rendertype$state);
         }
 
         public static Function<ResourceLocation, RenderType> TRANSLUCENT_PARTICLES_TARGET = Util.memoize(Internal::getTranslucentParticlesTarget);
 
         private static RenderType getTranslucentParticlesTarget(ResourceLocation locationIn) {
-            RenderType.CompositeState rendertype$state = RenderType.CompositeState.builder()
-                    .setShaderState(RENDERTYPE_TRANSLUCENT_SHADER)
+            var rendertype$state = RenderType.CompositeState.builder()
+                    .setShaderState(RenderType.RENDERTYPE_TRANSLUCENT_SHADER)
                     .setTextureState(new RenderStateShard.TextureStateShard(locationIn, false, true))
-                    .setTransparencyState(TRANSLUCENT_TRANSPARENCY)
-                    .setLightmapState(LIGHTMAP)
-                    .setOutputState(PARTICLES_TARGET)
+                    .setTransparencyState(RenderType.TRANSLUCENT_TRANSPARENCY)
+                    .setLightmapState(RenderType.LIGHTMAP)
+                    .setOutputState(RenderType.PARTICLES_TARGET)
                     .createCompositeState(true);
-            return create("neoforge_translucent_particles_target", DefaultVertexFormat.BLOCK, VertexFormat.Mode.QUADS, 2097152, true, true, rendertype$state);
+            return RenderType.create("neoforge_translucent_particles_target", DefaultVertexFormat.BLOCK, VertexFormat.Mode.QUADS, 2097152, true, true, rendertype$state);
         }
     }
 
     private static class CustomizableTextureState extends TextureStateShard {
+        private final Supplier<Boolean> blurSupplier;
+        private final Supplier<Boolean> mipmapSupplier;
+
         private CustomizableTextureState(ResourceLocation resLoc, Supplier<Boolean> blur, Supplier<Boolean> mipmap) {
```

Comment:
More of a nitpick, but if we're touching all this anyway, then we might as well change this to use a `BooleanSupplier` instead of throwing around boxed `Boolean`s

Reply:
I was actually wondering the same thing tbh but thought better to leave as original implementation had them, but I can quickly swap them out